src/asqi/workflow.py

```diff
@@ -865,6 +869,31 @@ def save_results_to_file_step(results: Dict[str, Any], output_path: str) -> None
         DBOS.logger.error(f"Invalid results data cannot be saved to {output_path}: {e}")
 
 
+@DBOS.step()
+def save_container_results_to_file_step(
```

Comment:
We can define `save_container_results_to_file_step` as a normal function instead of a DBOS step since it does not seem to be part of the workflow. Similarly, the previous `save_results_to_file_step` also seems to be wrongly configured as a DBOS step. 

I think we can treat both as side effects that does not need to be in the workflow